bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -220,6 +240,57 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Add an additional relayer tip for a committed message identified by `message_id`.
+		/// The tip asset will be swapped for ether.
+		#[pallet::call_index(2)]
+		#[pallet::weight(
+			T::WeightInfo::add_tip()
+				.saturating_add(T::BackendWeightInfo::transact_add_tip())
+		)]
+		pub fn add_tip(origin: OriginFor<T>, message_id: MessageId, asset: Asset) -> DispatchResult
+		where
+			<T as frame_system::Config>::AccountId: Into<Location>,
+		{
+			let who = ensure_signed(origin)?;
+
+			let ether_location = T::EthereumLocation::get();
+			let (tip_asset_location, tip_amount) = match asset {
+				Asset { id: AssetId(ref loc), fun: Fungibility::Fungible(amount) } => (loc, amount),
+				_ => return Err(Error::<T>::UnsupportedAsset.into()),
+			};
+
+			let ether_gained = if *tip_asset_location != ether_location {
+				Self::swap_and_burn(
+					who.clone(),
+					tip_asset_location.clone(),
+					ether_location,
+					tip_amount,
+				)
+				.map_err(|_| Error::<T>::SwapError)?
```

Comment:
Maybe the error itself is not that useful, but I would still grab it and log it :)

Reply:
Yes they are worthwhile. Any failing extrinsic on the live chain can be replayed under chopsticks using debug runtime with logs enabled. Massively helps debugging.

Reply:
Cool will add it!

Reply:
Added in https://github.com/paritytech/polkadot-sdk/pull/8271/commits/5bb4ffffe36527c6e2addb31fed55101e81609fc.